hera_mc/sys_handling.py

```diff
@@ -314,3 +314,17 @@ def publish_summary(self, hlist=['HH'], rev='A', exact_match=False,
             return 'OK'
         else:
             return 'Not on "main"'
+
+    def system_comments(self, system_kw='System', kword='all', tabkw=7):
+        longest_comment = 0
+        s = ''
+        for P in self.session.query(part_connect.PartInfo).filter((func.upper(part_connect.PartInfo.hpn) == system_kw.upper())):
+            if kword.lower() == 'all' or P.hpn_rev.lower() == kword.lower():
+                commlib = '{}  <{}>'.format(P.comment, P.library_file)
+                display_time = cm_utils.get_time_for_display(P.posting_gpstime)
+                s += "{:{tkw}s} | {:{pt}s} | {}\n".format(P.hpn_rev, display_time, commlib, tkw=tabkw, pt=len(display_time))
+                if len(commlib) > longest_comment:
+                    longest_comment = len(commlib)
+        header = "\n{:{tkw}s} | {:{pt}s} | {}  <{}>\n".format('Keyword', 'Posting', 'Comment', 'file/url', tkw=tabkw, pt=len(display_time))
+        header += "{}+{}+{}\n".format((tabkw + 1) * '-', (len(display_time) + 2) * '-', (longest_comment + 1) * '-')
```

Comment:
This doesn't *need* to be changed, but it's more efficient to build up a list of strings and then convert them into a multi-line string all in one go with `'\n'.join(line_list)`. Repeatedly extending long strings, as is done here, is an operation that can get quite slow.